enzyme/Enzyme/Enzyme.cpp

```diff
@@ -2824,7 +2937,10 @@ llvmGetPassPluginInfo() {
 #endif
               FunctionPassManager OptimizerPM;
               FunctionPassManager OptimizerPM2;
-#if LLVM_VERSION_MAJOR >= 14
+#if LLVM_VERSION_MAJOR >= 16
+              OptimizerPM.addPass(llvm::GVNPass());
+              OptimizerPM.addPass(llvm::SROAPass(llvm::SROAOptions::PreserveCFG));
```

Comment:
@wsmoses Not sure what setting should be passed here: ModifyCFG/PreserveCFG, could you please check?